code/modules/mob/living/simple_animal/hostile/hivebot.dm

```diff
@@ -31,85 +68,207 @@
 	playsound(src.loc, 'sound/effects/EMPulse.ogg', 25, 1)
 
-/mob/living/simple_animal/hostile/hivebot/tele/proc/warpbots()
+/mob/living/simple_animal/hostile/hivebotbeacon/death()
+	..(null,"blows apart and erupts in a cloud of noxious smoke!")
+	var/datum/effect/effect/system/smoke_spread/S = new /datum/effect/effect/system/smoke_spread
+	S.set_up(5, 0, src.loc)
+	S.start()
+	new /obj/effect/decal/cleanable/greenglow(src.loc)
+	var/T = get_turf(src)
+	new /obj/effect/gibspawner/robot(T)
+	spark(T, 3, alldirs)
+	for (var/mob/living/simple_animal/hostile/hivebot/latest_child in linked_bots)
+		latest_child.linked_parent = null
+	linked_bots.Cut()
+	qdel(src)
+	return
+
+/mob/living/simple_animal/hostile/hivebotbeacon/think()
+	. =..()
+	if(stance != HOSTILE_STANCE_IDLE && activated == 0)
+		visible_message("<span class='warning'>\ [src] suddenly activates!</span>")
+		icon_state = "hivebotbeacon_raising"
+		spawn(16)
+		icon_state = "hivebotbeacon_active"
+		activated = 1
+		snoozing = 0
+	else
+		if(activated == 1)
+			icon_state = "hivebotbeacon_active"
+
+/mob/living/simple_animal/hostile/hivebotbeacon/emp_act(severity)
+	LoseTarget()
+	stance = HOSTILE_STANCE_TIRED
+	addtimer(CALLBACK(src, .proc/wakeup), 200)
+	if(severity == 1.0)
+		apply_damage(80)
+	icon_state = "hivebotbeacon_off"
+	activated = -1
+
+/mob/living/simple_animal/hostile/hivebotbeacon/proc/wakeup()
+	stance = HOSTILE_STANCE_IDLE
+	activated = 0
 
+/mob/living/simple_animal/hostile/hivebotbeacon/proc/warpbots()
 	if(!bot_amt)
+		visible_message("<b>The Beacon</b> disappears in a cloud of smoke!")
+		playsound(src.loc, 'sound/effects/teleport.ogg', 25, 1)
+		var/datum/effect/effect/system/smoke_spread/S = new /datum/effect/effect/system/smoke_spread()
+		S.set_up(5, 0, src.loc)
+		S.start()
+		for (var/mob/living/simple_animal/hostile/hivebot/latest_child in linked_bots)
+			latest_child.linked_parent = null
+		linked_bots.Cut()
+		new /obj/effect/decal/cleanable/greenglow(src.loc)
 		qdel(src)
 		return
 
-	icon_state = "def_radar"
-	visible_message("<span class='warning'>\The [src] turns on!</span>")
-
-	switch(bot_type)
-		if(NORMAL)
-			new /mob/living/simple_animal/hostile/hivebot(get_turf(src))
-		if(RANGED)
-			new /mob/living/simple_animal/hostile/hivebot/range(get_turf(src))
-		if(RAPID)
-			new /mob/living/simple_animal/hostile/hivebot/range/rapid(get_turf(src))
-
-	if(prob(30))
-		if(prob(20))
-			bot_type = RAPID
+	if(activated != 1)
+		if(activated == -1)
+			return
 		else
-			bot_type = RANGED
-	else
-		bot_type = NORMAL
-	bot_amt--
-	addtimer(CALLBACK(src, .proc/warpbots, spawn_delay))
+			icon_state = "hivebotbeacon_raising"
+			sleep(16)
+			icon_state = "hivebotbeacon_active"
+			sleep(4)
+			activated = 1
 
+	if(existing_bots < max_bots)
+		visible_message("<span class='warning'>\ [src] radiates with energy!</span>")
+		switch(bot_type)
+			if(NORMAL)
+				latest_child = new /mob/living/simple_animal/hostile/hivebot(get_turf(src), src)

```

Comment:
`latest_child` doesn't have to be an instance variable, since it's only used in one proc. Just declare it locally on line 248:
```dm
		var/mob/living/simple_animal/hostile/hivebot/latest_child
		switch(bot_type)
			// whatever followed.
```